File: 1927-sum-game/1927-sum-game.java
class Solution {
    public boolean sumGame(String num) {
        int n = num.length();
        int half = n / 2;

        int leftSum = 0;
        int rightSum = 0;

        int qLeft = 0;
        int qRight = 0;

        for (int i = 0; i < half; i++) {
            char c = num.charAt(i);

            if (c == '?')
                qLeft++;
            else
                leftSum += c - '0';
        }

        for (int i = half; i < n; i++) {
            char c = num.charAt(i);

            if (c == '?')
                qRight++;
            else
                rightSum += c - '0';
        }

        int diff = leftSum - rightSum;

        return 2 * diff != 9 * (qRight - qLeft);
    }
}

File: 1927-sum-game/1927-sum-game.py
class Solution:
    def sumGame(self, num: str) -> bool:
        n = len(num)
        half = n // 2

        leftSum = 0
        rightSum = 0

        qLeft = 0
        qRight = 0

        for i in range(half):
            c = num[i]

            if c == '?':
                qLeft += 1
            else:
                leftSum += ord(c) - ord('0')

        for i in range(half, n):
            c = num[i]

            if c == '?':
                qRight += 1
            else:
                rightSum += ord(c) - ord('0')

        diff = leftSum - rightSum

        return 2 * diff != 9 * (qRight - qLeft)
